keychain.h

```diff
@@ -0,0 +1,70 @@
+#include <assert.h>
+
+#include <optional>
+#include <string>
+#include <variant>
+
+namespace keychain {
+
+class Keychain {
+   public:
+    struct Error {
+        std::string message;
+    };
+    using Success = std::monostate;
+    using Password = std::string;
+    using Result = std::variant<Success, Password, Error>;
+
+    static Result getPassword(const std::string &service,
+                              const std::string &user);
+    static Result setPassword(const std::string &service,
+                              const std::string &user,
+                              const std::string &password);
+    static Result deletePassword(const std::string &service,
+                                 const std::string &user);
+
+   public:
+    /*! \brief Construct a Keychain specific to a service/user combination
+     *
+     * \param service Identifier for the service whose password should be stored
+     *                or updated. Note that a platform specific service name
+     *                will be used based on the provided identifier.
+     *                \sa getServiceName
+     * \param user
+     */
+    Keychain(const std::string &service, const std::string &user);
```

Comment:
Revise parameters and which values are fixed by each instance. Introduce an additional service identifier that will be mangled to conform to platform specific requirements/conventions, possible provide service and/or user per call, so one instance can manage multiple passwords.